### DISCUSSION TITLE
How to be sure that everything is working fine?

### DISCUSSION THREAD

**Discussion by jvxis** [onemorebsmith/kaspa-stratum-bridge#41] [Q&A] [answered]
Hi @onemorebsmith I'm running for more than 18 hours, I ran all services with docker following the instructions here, my workers are all connected and sending shares. I'm using lolminer 1.59, and in total 22ghs. I don't hit a single block so far.

When I was mining directly into the node using bzminer, I hit 10 blocks in 3 days with 12ghs of power.

Am I doing something wrong?

Thanks,
JVX

**Comment by onemorebsmith** [comment]
lolminer 1.59 or 1.59a? You're also the 2nd person to report the bzminer issues, I'll take a look when I have time (currently on vacation) try dropping to a lower version. 

I hit 6 blocks with lolminer 1.59 over the last 24 hours with ~9GH/s so that should be fine.

**Reply by onemorebsmith** [reply]
Can you also check your miner address and see if you actually did get paid and not reported?

**Reply by jvxis** [reply]
lolminer 1.59. Very strange. Almost 24 hours and no blocks at all. I checked balances, and unfortunately, nothing there.
![image](https://user-images.githubusercontent.com/108929149/193049861-6e72518c-6622-4651-8ebd-5453a4dedd9e.png)

**Comment by onemorebsmith** [comment]
Yeah something definitely seems off on that. You can log in to prom and
check the hidden stats if you want. All stats start with `ks_` and there
should be an invalid share counter or something like that that might be
helpful.

You did also upgrade your node to 12.7, correct?

**Comment by jvxis** [comment]
Yes, I'm on 12.7.
Here is all the stats:
root@vps41575:~/kaspa-stratum-bridge-1.1.3# curl http://localhost:2114/metrics | grep ks_
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0# HELP ks_balance_by_wallet_gauge Gauge representing the wallet balance for connected workers
# TYPE ks_balance_by_wallet_gauge gauge
ks_balance_by_wallet_gauge{wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq"} 3094.68489429
ks_balance_by_wallet_gauge{wallet="kaspa:qqa2p6799zvfcxu24y3fl2u9285mmc5zrw2l4pac4m3msywjuf3qxda2zw2fs"} 1048.68549429
# HELP ks_estimated_network_hashrate_gauge Gauge representing the estimated network hashrate
# TYPE ks_estimated_network_hashrate_gauge gauge
ks_estimated_network_hashrate_gauge 2.56554623715365e+14
# HELP ks_invalid_share_counter Number of stale shares found by worker over time
# TYPE ks_invalid_share_counter counter
ks_invalid_share_counter{ip="179.100.20.25",miner="lolMiner 1.59",type="invalid",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv2"} 1
ks_invalid_share_counter{ip="179.100.20.25",miner="lolMiner 1.59",type="invalid",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv3"} 2
ks_invalid_share_counter{ip="187.111.227.45",miner="lolMiner 1.59",type="invalid",wallet="kaspa:qqa2p6799zvfcxu24y3fl2u9285mmc5zrw2l4pac4m3msywjuf3qxda2zw2fs",worker="Cropping_02"} 1
# HELP ks_network_block_count Gauge representing the network block count
# TYPE ks_network_block_count gauge
ks_network_block_count 225574
# HELP ks_network_difficulty_gauge Gauge representing the network difficulty
# TYPE ks_network_difficulty_gauge gauge
ks_network_difficulty_gauge 1.3177751795712045e+14
# HELP ks_valid_share_counter Number of shares found by worker over time
# TYPE ks_valid_share_counter counter
ks_valid_share_counter{ip="179.100.20.25",miner="BzMiner-v11.1.0",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv1"} 1789
ks_valid_share_counter{ip="179.100.20.25",miner="BzMiner-v11.1.0",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv2"} 18
ks_valid_share_counter{ip="179.100.20.25",miner="BzMiner-v11.1.0",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv3"} 1279
ks_valid_share_counter{ip="179.100.20.25",miner="BzMiner-v11.1.0",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv4"} 356
ks_valid_share_counter{ip="179.100.20.25",miner="SRBMiner-MULTI/1.0.8",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv2"} 8
ks_valid_share_counter{ip="179.100.20.25",miner="SRBMiner-MULTI/1.0.9",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv2"} 15
ks_valid_share_counter{ip="179.100.20.25",miner="lolMiner 1.59",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv1"} 9301
ks_valid_share_counter{ip="179.100.20.25",miner="lolMiner 1.59",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv2"} 10980
ks_valid_share_counter{ip="179.100.20.25",miner="lolMiner 1.59",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv3"} 14604
ks_valid_share_counter{ip="179.100.20.25",miner="lolMiner 1.59",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv4"} 6763
ks_valid_share_counter{ip="187.111.227.45",miner="lolMiner 1.59",wallet="kaspa:qqa2p6799zvfcxu24y3fl2u9285mmc5zrw2l4pac4m3msywjuf3qxda2zw2fs",worker="Cropping_01"} 14249
ks_valid_share_counter{ip="187.111.227.45",miner="lolMiner 1.59",wallet="kaspa:qqa2p6799zvfcxu24y3fl2u9285mmc5zrw2l4pac4m3msywjuf3qxda2zw2fs",worker="Cropping_02"} 6155
ks_valid_share_counter{ip="187.111.227.45",miner="lolMiner 1.59",wallet="kaspa:qqa2p6799zvfcxu24y3fl2u9285mmc5zrw2l4pac4m3msywjuf3qxda2zw2fs",worker="Cropping_03"} 17083
# HELP ks_worker_disconnect_counter Number of disconnects by worker
# TYPE ks_worker_disconnect_counter counter
ks_worker_disconnect_counter{ip="179.100.20.25",miner="BzMiner-v11.1.0",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv1"} 1
ks_worker_disconnect_counter{ip="179.100.20.25",miner="BzMiner-v11.1.0",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv2"} 1
10ks_worker_disconnect_counter{ip="179.100.20.25",miner="BzMiner-v11.1.0",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv3"} 1
0 ks_worker_disconnect_counter{ip="179.100.20.25",miner="BzMiner-v11.1.0",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv4"} 2
14ks_worker_disconnect_counter{ip="179.100.20.25",miner="SRBMiner-MULTI/1.0.8",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv2"} 1
62ks_worker_disconnect_counter{ip="179.100.20.25",miner="SRBMiner-MULTI/1.0.9",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv2"} 1
2  ks_worker_disconnect_counter{ip="179.100.20.25",miner="lolMiner 1.59",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv2"} 1
  0 ks_worker_disconnect_counter{ip="179.100.20.25",miner="lolMiner 1.59",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv3"} 2
146ks_worker_disconnect_counter{ip="179.100.20.25",miner="lolMiner 1.59",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv4"} 2
ks_worker_disconnect_counter{ip="187.111.227.45",miner="lolMiner 1.59",wallet="kaspa:qqa2p6799zvfcxu24y3fl2u9285mmc5zrw2l4pac4m3msywjuf3qxda2zw2fs",worker="Cropping_01"} 1
22ks_worker_disconnect_counter{ip="187.111.227.45",miner="lolMiner 1.59",wallet="kaspa:qqa2p6799zvfcxu24y3fl2u9285mmc5zrw2l4pac4m3msywjuf3qxda2zw2fs",worker="Cropping_02"} 2
  ks_worker_disconnect_counter{ip="187.111.227.45",miner="lolMiner 1.59",wallet="kaspa:qqa2p6799zvfcxu24y3fl2u9285mmc5zrw2l4pac4m3msywjuf3qxda2zw2fs",worker="Cropping_03"} 2
  # HELP ks_worker_errors Gauge representing errors by worker
0 # TYPE ks_worker_errors counter
  ks_worker_errors{error="err_failed_block_fetch",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq"} 3
  ks_worker_errors{error="err_failed_block_fetch",wallet="kaspa:qqa2p6799zvfcxu24y3fl2u9285mmc5zrw2l4pac4m3msywjuf3qxda2zw2fs"} 2
0 # HELP ks_worker_job_counter Number of jobs sent to the miner by worker over time
 # TYPE ks_worker_job_counter counter
3ks_worker_job_counter{ip="179.100.20.25",miner="BzMiner-v11.1.0",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv1"} 17785
56ks_worker_job_counter{ip="179.100.20.25",miner="BzMiner-v11.1.0",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv2"} 27
9ks_worker_job_counter{ip="179.100.20.25",miner="BzMiner-v11.1.0",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv3"} 8284
k ks_worker_job_counter{ip="179.100.20.25",miner="BzMiner-v11.1.0",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv4"} 3827
  ks_worker_job_counter{ip="179.100.20.25",miner="SRBMiner-MULTI/1.0.8",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv2"} 156
  ks_worker_job_counter{ip="179.100.20.25",miner="SRBMiner-MULTI/1.0.9",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv2"} 188
ks_worker_job_counter{ip="179.100.20.25",miner="lolMiner 1.59",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv1"} 8766
 0ks_worker_job_counter{ip="179.100.20.25",miner="lolMiner 1.59",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv2"} 3724
 --:--:-- --:--:-- --:--:-- 3569k
ks_worker_job_counter{ip="179.100.20.25",miner="lolMiner 1.59",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv3"} 4158
ks_worker_job_counter{ip="179.100.20.25",miner="lolMiner 1.59",wallet="kaspa:qpzz4383dh2evflp6wyjn9np9kdnkgx2c38jxqtcldmx07rd2t9jueyaan7xq",worker="rig-Jv4"} 96379
ks_worker_job_counter{ip="187.111.227.45",miner="lolMiner 1.59",wallet="kaspa:qqa2p6799zvfcxu24y3fl2u9285mmc5zrw2l4pac4m3msywjuf3qxda2zw2fs",worker="Cropping_01"} 200
ks_worker_job_counter{ip="187.111.227.45",miner="lolMiner 1.59",wallet="kaspa:qqa2p6799zvfcxu24y3fl2u9285mmc5zrw2l4pac4m3msywjuf3qxda2zw2fs",worker="Cropping_02"} 312
ks_worker_job_counter{ip="187.111.227.45",miner="lolMiner 1.59",wallet="kaspa:qqa2p6799zvfcxu24y3fl2u9285mmc5zrw2l4pac4m3msywjuf3qxda2zw2fs",worker="Cropping_03"} 114265

**Comment by jvxis** [comment]
Are you running yours with the Docker compose All? Or Are you running stratum bridge separated from the Monitoring.

**Reply by onemorebsmith** [reply]
`ks_invalid_share_counter` is 4 total. So for whatever reason you mined 4 blocks but they were rejected by the node. 

Possibly they were stale blocks but I don't have visibility into the reason at the moment.

**Reply by jvxis** [reply]
The problem might be related to the wallet I'm using?

I'm using tradeogre wallet, instead of kaspawallet, but I also used tradeogre wallet in my private node, which worked very well.

Any other clue? Maybe start the node again?

So the invalid shares that I got on miner and hiveos are actually rejected blocks?

Should I change the miner to bzminer? Or downgrade kaspa bridge version?

**Comment by onemorebsmith** [comment]
@jvxis https://github.com/onemorebsmith/kaspa-stratum-bridge/releases/tag/v1.1.4

**Comment by jvxis** [comment]
Hi @onemorebsmith. I will test this new version. I did the following test:
First, I ran my workers connected directly to the node using bzminer. I found 2 blocks and I got paid. So, my node is ok. Then, I setup again the kaspa-bridge, and connected my workers using lolminer 1.59. In total, more than 16 hours of running. I checked it now, and no blocks so far; and this time, no invalid share message, but I have an error message "failed_block_fetch":
![image](https://user-images.githubusercontent.com/108929149/193235403-f4f63658-33a2-4285-8f85-49bc88db8400.png)

![image](https://user-images.githubusercontent.com/108929149/193235253-59214851-8aaf-49b2-830b-cfbf2d1eee0b.png)

Do you know what this error message means? In the last log that I pasted here, I got 5 of them.

Thanks,
JVX

**Reply by jvxis** [reply]
Just adding. I'm running stratum bridge version 1.1.3 separated from the monitoring. All of my ports: 3000, 2114 and 9090 are opened so you can access my stats through URL kasbr.friendspool.club

**Reply by jvxis** [reply]
Running version 1.1.4 now, 12ghs connected with lolminer 1.59. Uptime 1h29m
![image](https://user-images.githubusercontent.com/108929149/193255488-2f098d46-d9e8-4ef5-a81b-6d875286463a.png)
This blank line with a different count appears in this new version.

**Reply by onemorebsmith** [reply]
The `failed block fetch` errors typically happen if your mining address is formatted incorrectly, if you lose connection to kaspad, or if kaspad loses sync

**Comment by jvxis** [comment, marked as answer]
Hi @onemorebsmith. Sorry, I want to bother you during your vacation; I'm just recording my findings here. No need to reply ASAP at all.
Did you check this in your Stratum after the hard fork?
![image](https://user-images.githubusercontent.com/108929149/193256874-f4a92cc6-0b30-4bc8-a250-431e63544c62.png)

**Reply by onemorebsmith** [reply]
I use the tx hash function from kaspad, it is updated

**Comment by onemorebsmith** [comment]
@jvxis keep me updated on how the new version is working! It should have addressed most of the issues you've found (hopefully)

**Reply by jvxis** [reply]
I will. Thanks. It is running, but no blocks yet. I changed the wallet address to this one: kaspa:qz3xfcqgqhmqvr0uyh783ss44k20er8cuvf2ktd89dnqpus82l5hck4207gl7, that I created using the web wallet app. What is weird: I can't see balances using this wallet, but I can see balances from the one on tradeogre.
until now, no errors. http://kasbr.friendspool.club:2114/metrics
![image](https://user-images.githubusercontent.com/108929149/193298080-5d45fbd6-9993-49f7-90f0-baa8930cca3b.png)

thanks,
JVX

**Comment by jvxis** [comment]
I'm still running...
![image](https://user-images.githubusercontent.com/108929149/193314456-3c54a30c-d5a5-4ccf-9b06-ed19e88cd037.png)
No errors so far, and unfortunately no blocks. I saw that you hit one block. Do you think the problem can be the miner sw? I've never found a block with another miner than bzminer, and I'm running the test 100% with lolminer 1.59.  The block that you found, Was it lolminer or bzminer?

**Reply by myntfarm** [reply]
ive been running since last night and ive gotten 4 blocks with lolminer 1.59a

**Reply by jvxis** [reply]
> ive been running since last night and ive gotten 4 blocks with lolminer 1.59a

Thanks. Good to know! Are you running version 1.1.4? How many Ghs? After all, it is just bad luck.

**Reply by myntfarm** [reply]
yes v1.1.4 or i pull directly from the git and docker compose it.
18 ghs

**Reply by jvxis** [reply]
> yes v1.1.4 or i pull directly from the git and docker compose it. 18 ghs

Thanks again.

**Reply by jvxis** [reply]
@Myntfarm which wallet are you using?

**Comment by myntfarm** [comment]
I’ve been using the web wallet from kaspanet 

—
Myntfarm
Vincent Place - Owner
(657) 215-5808


Friday, September 30, 2022, 10:13 AM -0700 from ***@***.***  ***@***.***>:
***@***.*** which wallet are you using?

**Reply by jvxis** [reply]
Are you using the monitoring? For the address from Web Wallet, I can't see balances on the monitoring dashboard. For the kaspawallet CLI wallet and tradeogre It is working.

**Comment by myntfarm** [comment]
Yes. I have no problems. I was confused at first with the rotating wallet addresses. But once I nailed one address down it all came together.

—
Myntfarm
Vincent Place - Owner
(657) 215-5808


Friday, September 30, 2022, 10:23 AM -0700 from ***@***.***  ***@***.***>:

**Reply by jvxis** [reply]
Thanks!

**Comment by jvxis** [comment]
![image](https://user-images.githubusercontent.com/108929149/193332992-7ca68877-59a4-490b-9eaf-696a205d18fb.png)
My IP Address is appearing now.

**Comment by jvxis** [comment]
![image](https://user-images.githubusercontent.com/108929149/193348344-abb606cd-75a0-4e89-a69a-fa6246794c37.png)
Finally! lolminer 1.59

**Reply by onemorebsmith** [reply]
Everything going well now?

**Reply by jvxis** [reply]
Yes. It is working very well. But I don't know why my IP address appears on the workers list in ks_bridge.

**Comment by onemorebsmith** [comment]
Make sure your worker name is set. Should be `Template%WAL%.%WORKER_NAME`
in hive

**Reply by jvxis** [reply]
![image](https://user-images.githubusercontent.com/108929149/193378533-4f09a581-6688-4fd5-97cb-79ed189a21b9.png)
Yes it is.

**Reply by jvxis** [reply]
![image](https://user-images.githubusercontent.com/108929149/193378584-cfb56702-af2a-4c76-b37f-e60886a8578a.png)
There is a blank line and then my IP Address. I also updated to version 1.1.4, but it is showing 1.1

**Reply by myntfarm** [reply]
@jvxis is it possible to try setting the worker name in the top right box
![image](https://user-images.githubusercontent.com/96462665/193379445-27ab3df0-342d-4433-8ab5-58c463c3696b.png)

**Reply by jvxis** [reply]
Thanks. I think it was caused by a occasional problem when I lost internet connection. Nothing related with flightsheet config. Mine is equal yours.

**Comment by jvxis** [comment]
Hi @onemorebsmith.
Running for 25 hours, 5 blocks. No problems.
![image](https://user-images.githubusercontent.com/108929149/193407936-6a51fe74-3bd0-43a8-b59f-0fc668ef2501.png)
Thanks for the support and great job done. If you need anything pls count on me.
